### PR TITLE
[NavigationDrawer] Fix issue showing the scrim when drawer scrolls too much

### PR DESCRIPTION
**Pull request by codeman7**
### The problem
The scrim is shown when a user scrolls past the content height.

### The fix
When a user scrolls add height to the content view controller.

### Related bugs
#5782 

### Videos
| Before | After |
| - | - |
|before|after|

**_Example snippet_**
```diff
// Within BottomDrawerWithHeaderLessContentExample
@objc override func presentNavigationDrawer() {
-   contentViewController.preferredHeight = 500
+   contentViewController.preferredHeight = 50
    super.presentNavigationDrawer()
  }
```

### CODE REVIEW

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m [hunk 1]
```diff
@@ -587,6 +587,27 @@ - (void)updateTopHeaderBottomShadowWithContentOffset:(CGPoint)contentOffset {
...
+  CGFloat previousAddedHeight = self.addedHeight;
+  self.addedHeight = height;
+  CGFloat heightToAdd = self.addedHeight - previousAddedHeight;
+  if (self.contentViewController) {
+    CGRect contentViewFrame = CGRectStandardize(self.contentViewController.view.frame);
+    contentViewFrame.size =
+        CGSizeMake(contentViewFrame.size.width, contentViewFrame.size.height + heightToAdd);
+    self.contentViewController.view.frame = contentViewFrame;
+  } else if (self.headerViewController) {
+    CGRect headerViewFrame = CGRectStandardize(self.headerViewController.view.frame);
```
Comment (yarneo) [inline components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m]: we shouldn't support cases where there is no contentViewController given, at this time. Seems like an odd behavior which wasn't tested yet and isn't part of the spec.

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m [hunk 2]
```diff
@@ -448,7 +448,6 @@ - (void)viewWillLayoutSubviews {
       contentViewFrame.size.height += topAreaInsetForHeader;
     }
   } else {
-    contentViewFrame.size.height = _contentVCPreferredContentSizeHeightCached;
```
Comment (yarneo) [inline components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m]: why was this removed

File: components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m [hunk 3]
```diff
@@ -569,4 +569,15 @@ - (void)testContentOffsetY {
...
 
+- (void)testAddedHeightWithContentViewController {
+  // Given
+  CGFloat contentViewControllerHeight =
+      CGRectStandardize(self.fakeBottomDrawer.contentViewController.view.frame).size.height;
+  [self.fakeBottomDrawer.scrollView setContentOffset:CGPointMake(0, 100)];
+  XCTAssertEqualWithAccuracy(self.fakeBottomDrawer.addedHeight, contentViewControllerHeight, 0.001);
+}
+
+- (void)testAddedHeightWithHeaderViewController {
```
Comment (yarneo) [inline components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m]: test is empty

**Review by yarneo — COMMENTED** [review]
can you please test how this reacts with landscape as well, and also that the existing apis like scroll to top and dismissing and re-presenting aren't affected by this.

can you provide a gif showing the new behavior?

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m (hunk 1 above)
Comment (codeman7) [reply components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m]: Removed

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m (hunk 2 above)
Comment (codeman7) [reply components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m]: That was a mistake, it has been added back.

File: components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m (hunk 3 above)
Comment (codeman7) [reply components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m]: Those were WIP I just pushed before I went home.
